tmt/steps/execute/internal.py

```diff
@@ -396,12 +394,12 @@ def _handle_reboot(self, test: Test, guest: Guest) -> bool:
         REBOOTCOUNT variable, reset it to 0 if no reboot was requested
         (going forward to the next test). Return whether reboot was done.
         """
-        if self._will_reboot(test, guest):
-            test._reboot_count += 1
-            self.debug(f"Reboot during test '{test}' "
-                       f"with reboot count {test._reboot_count}.")
-            reboot_request_path = self._reboot_request_path(test, guest)
-            test_data = self.data_path(test, guest, full=True) / tmt.steps.execute.TEST_DATA
+        if self._will_reboot(invocation):
+            invocation.test._reboot_count += 1
```

Comment:
Isn't `_reboot_count` something which would fit to `invocation` as well?

Reply:
Exactly. The whole reboot handling would probably fit into `invocation` as well, but that's a big one. In this PR, I wanted to start the moving rather than complete it.

That said, the very next patch will deal with the following fields:

```python
    return_code: Optional[int] = field(default=None, internal=True)
    start_time: Optional[str] = field(default=None, internal=True)
    end_time: Optional[str] = field(default=None, internal=True)
    real_duration: Optional[str] = field(default=None, internal=True)
    _reboot_count: int = field(default=0, internal=True)
```

These are stored in `Test` instances, and `Test` instances might be shared among different threads running the same test on different guests in one phase. Overwriting the values as they finish, making results misleading...

Reply:
Yeah, this cleanup would be very good! So a separate patch it will be. Ack.